Formula/j/julia.rb

```diff
@@ -73,13 +74,13 @@ def install
       USE_BINARYBUILDER=0
       USE_SYSTEM_BLAS=1
       USE_SYSTEM_CSL=1
-      USE_SYSTEM_CURL=1
+      USE_SYSTEM_CURL=0
```

Comment:
```suggestion
      USE_SYSTEM_CURL=#{OS.mac? ? "1" : "0"}
```
We can probably still use macOS `libcurl` here.

Reply:
Although it builds/tests, `Downloads.Curl` still spits out errors messages whenever system curl is used.

Building with `USE_SYSTEM_CURL=0` is the same.

May need to try building a resource-ed `curl`.

Reply:
> May need to try building a resource-ed `curl`.

Ok with `USE_SYSTEM_CURL=0` instead if we have to resort to that.

Reply:
`USE_SYSTEM_CURL=0` doesn't seem to be working as it outputs the same errors and there is no `libcurl` in bottle.

Reply:
Current run also lacks `libcurl.{so,dylib}`. Comparing to official binary distribution, they do have `julia-1.11.0-rc4/lib/julia/libcurl.so`

Reply:
Official binaries will set `USE_BINARY_BUILDER=1`, so `USE_BINARY_BUILDER=0` and `USE_SYSTEM_CURL=0` is probably very untested.

Reply:
It could be that build needs bundled MBedTLS, libssh2, and nghttp2: https://github.com/JuliaLang/julia/blob/v1.11.0-rc4/deps/curl.mk#L45

Anyway, I think running extra `make` command to build the bundled `curl` works.